WalletWasabi/Crypto/StringCipher.cs

```diff
@@ -1,114 +1,115 @@
-using System.IO;
 using System.Linq;
 using System.Security.Cryptography;
 using System.Text;
 using WalletWasabi.Crypto.Randomness;
 
 namespace WalletWasabi.Crypto
 {
-	// https://stackoverflow.com/a/10177020/2061103
 	public static class StringCipher
 	{
 		// This constant is used to determine the keysize of the encryption algorithm in bits.
 		// We divide this by 8 within the code below to get the equivalent number of bytes.
-		private const int KeySize = 128;
+		private const int KeyByteSize = 128 / 8;
+		private const int AesBlockByteSize = 128 / 8;
+		private const int PasswordSaltByteSize = 128 / 8;
+		private const int SignatureByteSize = 256 / 8;
 
 		// This constant determines the number of iterations for the password bytes generation function.
 		private const int DerivationIterations = 1000;
 
+		private const int MinimumEncryptedMessageByteSize =
+			PasswordSaltByteSize + // Auth salt
+			PasswordSaltByteSize + // Key salt
+			AesBlockByteSize + // IV
+			AesBlockByteSize + // Cipher text min length
+			SignatureByteSize; // Signature tag
+
+		private static readonly SecureRandom SecureRandom = new();
```

Comment:
`SecureRandom` is disposable.

Reply:
Possibly one can just do:

```cs
public static string Encrypt(string plainText, string passPhrase)
{
	using SecureRandom secureRandom = new();
	byte[] iv = secureRandom.GetBytes(AesBlockByteSize);
	byte[] encryptionKeySalt = secureRandom.GetBytes(PasswordSaltByteSize);
	byte[] encryptionKey = DerivateKey(passPhrase, encryptionKeySalt);

	// Encrypt the plain text
	using var aes = CreateAES(encryptionKey);
	byte[] plainTextBytes = Encoding.UTF8.GetBytes(plainText);
	byte[] cipherTextBytes = aes.EncryptCbc(plainTextBytes, iv, PaddingMode.PKCS7);

	// Authenticate
	byte[] authKeySalt = secureRandom.GetBytes(PasswordSaltByteSize);
	byte[] authKey = DerivateKey(passPhrase, authKeySalt);
	byte[] result = MergeArrays(additionalCapacity: SignatureByteSize, encryptionKeySalt, iv, authKeySalt, cipherTextBytes);

	byte[] authCode = HMACSHA256.HashData(authKey, result[..^SignatureByteSize]);
	authCode.CopyTo(result, result.Length - SignatureByteSize);

	return Convert.ToBase64String(result);
}
```

or pass `SecureRandom` to `GenerateRandomEntropy`.